src/test/rustdoc/remove-url-from-headings.rs

```diff
@@ -0,0 +1,17 @@
+#![crate_name = "foo"]
+
+// @has foo/fn.foo.html
+// !@has - '//a[@href="http://a.a"]'
+// @has - '//a[@href="#implementing-stuff-somewhere"]' 'Implementing stuff somewhere'
+// @has - '//a[@href="#another-one-urg"]' 'Another one urg'
+
+/// fooo
+///
+/// # Implementing [stuff](http://a.a) somewhere
```

Comment:
```suggestion
/// # Implementing [stuff](http://a.a "title") somewhere
```

We may as well check that the link title doesn't show up accidentally while we're at it.